### PR TITLE
* #87: Added error for role + group on the same table.

### LINKED ISSUE #87: Dangerous behavior for combination of role- and group-based security
Combining role- and group-based security for the same table is not supported, but if such a table is accidentally created, then only the roles get checked and the groups get ignored. Instead, no access should be possible or an error should be returned.

### PR DESCRIPTION
**Pull request by AnastasiiaSergienko**
Closes #87

### CODE REVIEW

File: src/main/java/com/exasol/adapter/dialects/rls/RowLevelSecurityQueryRewriter.java [hunk 1]
```diff
@@ -132,6 +132,9 @@ private SqlNode combineRegularFilterWithRlsFilter(final SqlStatementSelect selec
                 return and(where, createTenantsNode(userInformation));
             }
         } else if (protection.isRoleProtected()) {
+            if (protection.isGroupProtected()) {
+                throw getRoleAndGroupException();
+            }
             return and(where, createRolesNode(userInformation));
```
Comment (redcatbear) [inline src/main/java/com/exasol/adapter/dialects/rls/RowLevelSecurityQueryRewriter.java]:
```suggestion
		else {
            return and(where, createRolesNode(userInformation));
       }
```
Symmetry.

File: src/main/java/com/exasol/adapter/dialects/rls/RowLevelSecurityQueryRewriter.java [hunk 2]
```diff
@@ -222,6 +230,9 @@ private SqlNode createRlsFilter(final UserInformation userInformation, final Tab
                 return createTenantsNode(userInformation);
             }
         } else if (protection.isRoleProtected()) {
+            if (protection.isGroupProtected()) {
+                throw getRoleAndGroupException();
+            }
             return createRolesNode(userInformation);
```
Comment (redcatbear) [inline src/main/java/com/exasol/adapter/dialects/rls/RowLevelSecurityQueryRewriter.java]:
```suggestion
		else {
            return createRolesNode(userInformation);
       }
```

File: src/test/java/com/exasol/adapter/dialects/rls/AbstractRowLevelSecurityIT.java [hunk 3]
```diff
@@ -204,6 +205,31 @@ void testGroupRestrictedTableWithUnauthorizedUser() throws SQLException {
         assertThat(exception.getMessage(), containsString("E-VS-RLS-JAVA-7"));
     }
 
+    @Test
+    void testGroupAndRoleRestrictedTable() {
```
Comment (redcatbear) [inline src/test/java/com/exasol/adapter/dialects/rls/AbstractRowLevelSecurityIT.java:209]: Two test cases in one method.

**Review by redcatbear — CHANGES_REQUESTED** [review]

File: src/test/java/com/exasol/adapter/dialects/rls/AbstractRowLevelSecurityIT.java (hunk 3 above)
Comment (AnastasiiaSergienko) [reply src/test/java/com/exasol/adapter/dialects/rls/AbstractRowLevelSecurityIT.java:209]: I wanted to avoid copying the test, it's quite long

Comment (AnastasiiaSergienko) [reply src/test/java/com/exasol/adapter/dialects/rls/AbstractRowLevelSecurityIT.java:209]: Split them into 2 tests

**Review by redcatbear — DISMISSED** [review]

**Pull request merged**